src/video/tinspire/SDL_tinspirevideo.c

```diff
@@ -288,6 +288,17 @@ static int NSP_SetColors(_THIS, int firstcolor, int ncolors, SDL_Color *colors)
 */
 static void NSP_VideoQuit(_THIS)
 {
+	if (this->hidden->buffer != NULL)

```

Comment:
`free(NULL);`  is defined to be a noop, so the `if` can be removed

Reply:
So i should just free buffer2 ?

Reply:
Alright i did it the proper way but not sure what you meant by removing the if statement. I think you meant that i should have removed the != NULL instead.